.github/actions/test-source/action.yml

```
@@ -23,10 +23,14 @@ runs:
         restore-keys: |
           ${{ runner.os }}-gradle-
 
+    - name: Run build to check if there are any checkstyle issues
```

```suggestion
    - name: Build project
```

This will also result in a fail if there are any compilation errors